### PR TITLE
Add first api endpoint for pull requests

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

**Comment by democratic-collaboration** [comment]
Adding WIP label, the title is prefixed with [WIP]

**Comment by democratic-collaboration** [comment]
Adding WIP label, the title is prefixed with [WIP]

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Code update recognized, countdown starts fresh.

**Comment by democratic-collaboration** [comment]
Removing WIP label, the title is not prefixed with [WIP]

File: src/api.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from PullRequest import mergeable_pull_request
+import github
+from flask import g
+
+class APIPullRequest(restful.Resource):
+    def get(self, org, repo, pull):
+        if not g.user:
+            abort(401)
+            return
+        print(g.user)
```
Comment (pst) [inline src/api.py]: Print?

File: src/PullRequest.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    if commentOnIssue:
+        issue.create_comment(message)
+
+    if age >= days_to_merge:
+        print('Would merge now')
+        try:
+            pull_request.merge()
+        except Exception as e:
+            # Maybe add a comment that the conflicts should be resolved
+            print(e)
```
Comment (pst) [inline src/PullRequest.py]: Print?

File: src/PullRequest.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    # days_to_merge = (1 - coefficient) * calculated days
+    merge_duration = timedelta(days=(1 - coefficient) * (5 + pull_request.commits * 5))
+    days_to_merge = merge_duration - age
+    message = '''A new review, yeah.
+
+    Votes: {}/{}
+    Coefficient: {}
+    Merging in {} days {} hours
+    Age {} days {} hours'''.format(votes, votes_total, coefficient, days_to_merge.days, days_to_merge.seconds / 3600, age.days, age.seconds / 3600)
+    print(message)
```
Comment (pst) [inline src/PullRequest.py]: Print?

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 71/77
    Coefficient: 0.922077922078
    Merging in 0 days 18 hours
    Age 0 days 0 hours

File: src/api.py (hunk 1 above)
Comment (TooAngel) [inline src/api.py]: These parts are just copied from another file, I would change that later.

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 75/77
    Coefficient: 0.974025974026
    Merging in 0 days 6 hours
    Age 0 days 0 hours

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 76/77
    Coefficient: 0.987012987013
    Merging in 0 days 2 hours
    Age 0 days 0 hours

**Pull request merged**